src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tasks/GetDotNetHost.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace Microsoft.AspNetCore.BuildTools
+{
+    public class GetDotNetHost : Task
+    {
+        [Output]
+        public string MuxerPath { get; set; }
+
+        public override bool Execute()
+        {
+            MuxerPath = DotNetMuxer.MuxerPathOrDefault();
```

Comment:
Loosely based on https://github.com/aspnet/BuildTools/blob/dev/src/Internal.AspNetCore.BuildTools.Tasks/GetDotNetHost.cs. We'll fallback to using the exe from the path in the event the user is using desktop msbuild to publish a netcoreapp2.0 application.